### PR TITLE
srv6: Some cleanups for SRv6 maps

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
Please see individual commits for more details.

```release-note
srv6: Some cleanups for SRv6 maps
```

### CODE REVIEW

**Review by rgo3 — APPROVED** [review]
ACK for Loader changes

**Review by glibsm — APPROVED** [review]
Proto change looks good.

**Review by aspsk — APPROVED** [review]

**Review by derailed — APPROVED** [review]
@YutaroHayakawa Nice work!

File: pkg/maps/srv6map/policy.go [hunk 1]
```diff
@@ -182,3 +129,71 @@ func (m srv6PolicyMap) IterateWithCallback6(cb SRv6PolicyIterateCallback) error
...
+			"SRV6_POLICY_MAP_SIZE": strconv.FormatUint(maxPolicyEntries, 10),
+		},
+	}
+
+	return bpf.NewMapOut(&PolicyMap4{m4}), bpf.NewMapOut(&PolicyMap6{m6}), nodeOut
+}
+
+// OpenPolicyMaps opens the SRv6 policy maps on bpffs
+func OpenPolicyMaps() (*PolicyMap4, *PolicyMap4, error) {
+	m4, err := ebpf.LoadRegisterMap(policyMapName4)
```
Comment (ti-mo) [inline pkg/maps/srv6map/policy.go]: Nit: I know this isn't advertised anywhere prominently, but the internal ebpf package is deprecated. Unless there's a feature missing from package bpf, try to use that one instead. Sorry!

Comment (YutaroHayakawa) [reply pkg/maps/srv6map/policy.go]: Oops, ok let me take this opportunity and fix that.

**Comment by YutaroHayakawa** [comment]
Cilium E2E Upgrade: https://github.com/cilium/cilium/issues/30721
Conformance Ginkgo: Job timeout

File: pkg/maps/srv6map/policy.go (hunk 1 above)
Comment (YutaroHayakawa) [reply pkg/maps/srv6map/policy.go]: Reflected the suggestion.

**Review by ti-mo — APPROVED** [review]

File: pkg/maps/srv6map/state.go [hunk 2]
```diff
@@ -4,119 +4,18 @@
...
-				InnerDst: &dstIP,
-			}
-			value := v.(*StateValue)
-
-			cb(&key, value)
-		})
+// We can delete this in v1.18
+func cleanupStateMap() {
+	os.Remove(bpf.MapPath(StateMapName4))
+	os.Remove(bpf.MapPath(StateMapName6))
```
Comment (julianwiedmann) [inline pkg/maps/srv6map/state.go]: @YutaroHayakawa :wave: in case you still want to remove this

Comment (YutaroHayakawa) [reply pkg/maps/srv6map/state.go]: Thanks! Will remove this in this PR: https://github.com/cilium/cilium/pull/42184.

**Pull request merged**